### PR TITLE
[inference] add int8 rotary embedding kernel for smoothquant

### PR DESCRIPTION
**Pull request by Xu-Kai**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [ ] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?
- Add in8 rotary embedding kernel for smoothquant

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [ ] I have performed a self-review of my code
- [x] I have added thorough tests.
- [ ] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: tests/test_smoothquant/test_rotary_embedding.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    input_scale = torch.max(torch.abs(x)) / 127
+    output_scale = torch.max(torch.abs(y_torch)) / 127
+
+    x = x / input_scale
+    x = x.to(torch.int8)
+
+    int8_rotary_embedding_fwd(x, cos, sin, input_scale, output_scale)
+    y_triton = x.to(torch.float) * output_scale
+    assert torch.allclose(y_triton, y_torch, atol=2e-1, rtol=1e-2, equal_nan=True)
```
Comment (yuanheng-zhao) [inline tests/test_smoothquant/test_rotary_embedding.py:57]: Is this `atol=2e-1` too large?

File: tests/test_smoothquant/test_rotary_embedding.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# Adapted from ModelTC https://github.com/ModelTC/lightllm
+
+
+import pytest
+import torch
+from packaging import version
+
+try:
+    pass
```
Comment (yuanheng-zhao) [inline tests/test_smoothquant/test_rotary_embedding.py:9]: remove this

File: colossalai/kernel/triton/int8_rotary_embedding_kernel.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    o_scale = tl.load(output_scale)
+
+    q0 = q0.to(tl.float32) * in_scale
+    q1 = q1.to(tl.float32) * in_scale
+
+    out0 = (q0 * cos - q1 * sin) / o_scale
+    out1 = (q0 * sin + q1 * cos) / o_scale
+
+    # out0 = out0.to(tl.int8)
+    # out1 = out1.to(tl.int8)
```
Comment (yuanheng-zhao) [inline colossalai/kernel/triton/int8_rotary_embedding_kernel.py:70]: remove lines for testing

File: tests/test_smoothquant/test_rotary_embedding.py (hunk 1 above)
Comment (Xu-Kai) [reply tests/test_smoothquant/test_rotary_embedding.py:57]: can not pas to set smaller value.

File: colossalai/kernel/triton/int8_rotary_embedding_kernel.py (hunk 3 above)
Comment (Xu-Kai) [reply colossalai/kernel/triton/int8_rotary_embedding_kernel.py:70]: Removed.

File: tests/test_smoothquant/test_rotary_embedding.py (hunk 2 above)
Comment (Xu-Kai) [reply tests/test_smoothquant/test_rotary_embedding.py]: Removed.

**Review by yuanheng-zhao — APPROVED** [review]

**Pull request merged**